FirebaseRemoteConfig/Sources/RCNConfigRealtime.m

```diff
@@ -455,17 +484,27 @@ - (void)URLSession:(NSURLSession *)session
     didReceiveData:(NSData *)data {
   NSError *dataError;
   NSString *strData = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
-  strData = [strData substringFromIndex:1];
-  data = [strData dataUsingEncoding:NSUTF8StringEncoding];
-  NSDictionary *response = [NSJSONSerialization JSONObjectWithData:data
-                                                           options:NSJSONReadingMutableContainers
-                                                             error:&dataError];
-  NSString *targetTemplateVersion = _configFetch.templateVersionNumber;
-  if (dataError == nil) {
-    targetTemplateVersion = [response objectForKey:kTemplateVersionNumberKey];
+  NSRange endRange = [strData rangeOfString:@"}"];
+  NSRange beginRange = [strData rangeOfString:@"{"];
+  if (beginRange.location != NSNotFound && endRange.location != NSNotFound) {
+    FIRLogDebug(kFIRLoggerRemoteConfig, @"I-RCN000015",
+                @"Received config update message on stream.");
+    NSRange msgRange =
+        NSMakeRange(beginRange.location, endRange.location - beginRange.location + 1);
+    strData = [strData substringWithRange:msgRange];
+    data = [strData dataUsingEncoding:NSUTF8StringEncoding];
+    NSDictionary *response = [NSJSONSerialization JSONObjectWithData:data
+                                                             options:NSJSONReadingMutableContainers
+                                                               error:&dataError];
+    NSInteger oldTemplateVersion = [_configFetch.templateVersionNumber integerValue];
+    if (dataError == nil) {
+      NSInteger updateTemplateVersion =
+          [[response objectForKey:kTemplateVersionNumberKey] integerValue];
+      if (updateTemplateVersion > oldTemplateVersion) {
```

Comment:
so when an experiment is updated the template version does not increment. should this be ">=" to handle that case?

and should we add a check to not autofetch if "templateVersion" is not in the response?

Reply:
Changed to >= and added check for templateVersion in message